### PR TITLE
Introduce support for primitive types

### PR DESCRIPTION
**Pull request by fbiville**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 

## Description

This adds support for primitive types, particularly for custom change fields.

## Things to be aware of

It could potentially change the converted value type of existing changes, but that's unlikely as the modified function only returns primitive types if explicitly requested.
Since such requests would fail in the past, I doubt this change will have an impact.

## Things to worry about

Changing existing `Change` subclasses to migrate from boxed types to primitive types is likely a bad idea since it may change the generated check sum.
The general change proposed here is however useful for new changes.

## Additional Context

N/A

### CODE REVIEW

File: liquibase-standard/src/test/java/com/example/liquibase/change/ChangeWithPrimitiveFields.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import liquibase.change.DatabaseChange;
+import liquibase.database.Database;
+import liquibase.statement.SqlStatement;
+
+@DatabaseChange(
+        name = "primitiveChange",
+        description = "Used in unit tests",
+        priority = ChangeMetaData.PRIORITY_DEFAULT
+)
+public class ChangeWithPrimitiveFields extends AbstractChange {
```
[the remark this answers is not in the document]
Comment (fbiville) [reply liquibase-standard/src/test/java/com/example/liquibase/change/ChangeWithPrimitiveFields.java:14]: Is there to bypass this check? `liquibase.AbstractExtensibleObject#clone` seems to do the work already.

**Review by filipelautert — APPROVED** [review]

**Comment by rberezen** [comment]
Hi @fbiville , could you please take a look at the unit test failures in this link: https://github.com/liquibase/liquibase/actions/runs/7265842115/job/19796976554?pr=5360? Your assistance in resolving them would be greatly appreciated. Thank you!
cc @filipelautert

**Comment by MalloD12** [comment]
Hi @fbiville,

I have fixed some unit tests that were failing and made some changes to some tests that were not running all the scenarios specified in the `where` sections. One `StringChangeLogSerializerTest` test is failing only under Windows check. Would you mind taking a look and fixing that scenario?

Thanks,
Daniel.

**Comment by fbiville** [comment]
Hello, I'm back from vacation, I will have a look now.

**Comment by fbiville** [comment]
@MalloD12 I cannot reproduce the issue, the `ChangeWithPrimitiveFields` does not seem to be included in the changes tested by `StringChangeLogSerializerTest`

**Comment by MalloD12** [comment]
> Details

Hi @fbiville,

Same here, I cannot reproduce that issue locally either (but that's because I'm on a Mac) I'll ask someone from the team for some assistance trying to reproduce this issue. I do have not a clear understanding of why that test is failing, or what is the relationship between the type returned by `ChangeWithPrimitiveFields` and what's being tested in `StringChangeLogSerializerTest`. 

Thanks,
Daniel.

**Comment by fbiville** [comment]
As discussed on Slack, it turns the issue came from the fact that the test I added registered a custom change but did not unregister it when the test ended, which had the unwanted effect of keeping that specific change available for other tests.

**Review by MalloD12 — APPROVED** [review]
**Approved**.

Code changes look good to me. Thank you @fbiville for this PR and fixing up the last failing test.

Build and most of the test checks have been successfully executed, except for the functional test suite failing for a known issue when downloading artifacts.

**Review by rberezen — APPROVED** [review]

**Pull request merged**